src/types/struct.rs

```diff
@@ -233,6 +233,13 @@ impl Struct {
     pub fn get_all<A: AsSymbolRef>(&self, field_name: A) -> impl Iterator<Item = &Element> {
         self.fields.get_all(field_name)
     }
+
+    pub fn get_index(&self, field_index: usize) -> Option<(&Symbol, &Element)> {
```

Comment:
I'm skeptical about this being `pub` because Ion doesn't guarantee ordering of fields in a struct. Can it be `pub(crate)`, at least for now?